postgres_composite_types/__init__.py

```diff
@@ -225,6 +210,58 @@ def __new__(mcs, name, bases, attrs):
 
         return new_cls
 
+    def __init__(cls, name, bases, attrs):
+        super().__init__(name, bases, attrs)
+        if name == 'CompositeType':
+            return
+
+        # Register the type on the first database connection
+        connection_created.connect(receiver=cls.database_connected,
+                                   dispatch_uid=cls._meta.db_type)
+
+    def database_connected(cls, signal, sender, connection, **kwargs):
+        if isinstance(connection, PostgresDatabaseWrapper):
+            # Try to register the type. If the type has not been created in a
+            # migration, the registration should be retried later.
+            cls.register_composite(connection, retry=True)
+
+        # Disconnect the signal now - only need to register types on the
+        # initial connection
+        connection_created.disconnect(cls.database_connected,
+                                      dispatch_uid=cls._meta.db_type)
+
+    def register_composite(cls, connection, retry=False):
```

Comment:
Same as above, the registration needs to either happen as the application starts up, or after the type has been created. This method is called as part of both events.